Composer/packages/server/src/models/bot/botProject.ts

```diff
@@ -644,7 +644,6 @@ export class BotProject {
                 replacers.map(replacer => {
                   newLine = replacer(newLine);
                 });
-                newLine = newLine.replace('-', '_');
                 newContentLines.push(newLine);
```

Comment:
so the fix is just don't replace?

Reply:
Fixed by only replace items defined in replacers(above line), this line is too generic and risky. By remove this line, the replacement in *.dialog *.lg would be same. 
includes 
```
bfdactivity-1234 -> SendActivity_1234
bfdprompt-1234 -> TextInput_Prompt_1234
bfdinvalidPrompt-1234 -> TextInput_InvalidPrompt_1234
bfdunrecognizedPrompt-1234 -> TextInput_UnrecognizedPrompt_1234
bfddefaultValueResponse-1234 -> TextInput_DefaultValueResponse_1234
```
excludes described in #2597 

Won't auto-replace case:

reference is user customize defined lg template name,
- ${my-template()}
technically can not distinguish it is an expression or a template reference.

"